src/main/java/org/jenkinsci/plugins/electricflow/event/ElectricFlowBuildWatcher.java

```diff
@@ -182,5 +183,33 @@ public boolean sendBuildDetailsToInstanceImproved(Run<?, ?> run, TaskListener ta
     return true;
   }
 
+  /**
+   * CI builds triggered by CD primarily store related information in {@link EFCause}, whereas CD pipelines, releases,
+   * and attachments triggered by CI store related information in {@link CloudBeesCDPBABuildDetails}.
+   *
+   * <p>This method adds a {@link CloudBeesCDPBABuildDetails} action corresponding the the {@link EFCause} if one is
+   * present so that all CD-related associations can be viewed in the Jenkins REST API just by browsing the
+   * {@link CloudBeesCDPBABuildDetails} actions attached to a build.
+   *
+   * <p>{@link EFCause} takes priority in {@link #sendBuildDetailsToInstanceImproved} so this should not affect existing
+   * behavior.
+   */
+  private void addActionForBuildsTriggeredByCD(Run<?, ?> run) {
```

Comment:
Before this PR, there was a disconnect between `EFCause`, which is for builds triggered by CD, and `CloudBeesCDPBABuildDetails`, which was for CD associations triggered by CI. I think there is value in having all of that data be exposed to the Jenkins REST API using the same `CloudBeesCDPBABuildDetails` class since the overall structure of the data is pretty much the same.

The naming does become a little weird now though, since the `PBA` part of `CloudBeesCDPBABuildDetails` is not accurate for this case since there is no post-build action. If that is undesirable, maybe we could introduce a new view into the data that could be exposed to the REST API instead that would unify everything?